### PR TITLE
Bb refactor

### PR DESCRIPTION
**Pull request by GDYendell**

### CODE REVIEW

**Comment by GDYendell** [comment]
This is the GitServer update with GitoliteServer as the active class. I have updated everything so the system tests pass with this new structure. We should be able to make another pull request that enables the Bitbucket server and (along with a few additions, such as getting user from the environment) it will just (!) start using the Bitbucket server instead.

File: dls_ade/dls_logs_since_release.py [hunk 1]
```diff
@@ -118,21 +127,28 @@ def check_releases_valid(releases, parser):
...
-        parser.error("Only two releases can be specified (start and end point)")
+        parser.error("Only two releases can be specified (start and end"
+                     " point)")
     elif len(releases) == 2:
-        # If start and end points both provided then check that later comes after earlier
+        # If start and end points both provided then check that later comes
+        # after earlier
         env = environment()
         test_list = env.sortReleases([releases[0], releases[1]])
         if releases[1] == test_list[0] and releases[1] != 'HEAD':
```
Comment (nickbattam) [inline dls_ade/dls_logs_since_release.py]: This error doesn't make sense "you have supplied the releases in the correct order"

**Comment by ulrikpedersen** [comment]
Thanks @nickbattam for the comprehensive review! :+1:

**Comment by GDYendell** [comment]
I have made the changes suggested that were directly related to this.

There are a lot of things I want to change here, probably in multiple branches, so I will implement some of these suggestions there. I have a page for it [here](http://confluence.diamond.ac.uk/display/~mef65357/Notes+on+DLS+ADE).

**Pull request merged**